core/src/main/kotlin/com/exactpro/th2/read/db/bootstrap/Main.kt

```diff
@@ -187,19 +195,47 @@ internal fun setupApp(
     READINESS_MONITOR.enable()
 }
 
+private fun createMessageLoader(
+    factory: CommonFactory,
+    componentBookName: String
+) = runCatching {
+    MessageSearcher(factory.grpcRouter.getService(DataProviderService::class.java)).run {
+        object : MessageLoader {
+            override fun load(dataSourceId: DataSourceId, properties: Map<String, String>): TableRow? =
+                searchLastOrNull(
+                    book = componentBookName,
+                    sessionAlias = dataSourceId.id,
+                    direction = com.exactpro.th2.common.grpc.Direction.FIRST,
+                    interval = Duration.ofDays(1), // FIXME: use parameter for that
+                ) {
+                    properties.all { (key, value) -> it.message.getMessagePropertiesOrDefault(key, null) == value }
+                }?.message?.bodyRaw?.toByteArray()?.let {
+                    MAPPER.readValue(it, TableRow::class.java)
```

Comment:
Could you please advise on how exactly you are going to read a CSV raw message into TableRow using a MAPPER form CommonFactory?

Reply:
I was wrong